# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/SecFetch.cpp
# Commit: 55f13fb4ee3f
# Full Hash: 55f13fb4ee3f06209d00fd91a7a03884bab88e2d
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement csp for Module Workers; r=evilpie,asuth,rpl,ckerschb
#   
#   Depends on D155691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156102
# ==============================================================================

diff -r 354711cf113a -r 55f13fb4ee3f dom/security/SecFetch.cpp
--- a/dom/security/SecFetch.cpp	Wed Dec 14 14:55:37 2022 +0000
+++ b/dom/security/SecFetch.cpp	Wed Dec 14 14:55:38 2022 +0000
@@ -31,6 +31,7 @@
     case nsIContentPolicy::TYPE_SCRIPT:
       return "script"_ns;
     case nsIContentPolicy::TYPE_INTERNAL_WORKER:
+    case nsIContentPolicy::TYPE_INTERNAL_WORKER_STATIC_MODULE:
       return "worker"_ns;
     case nsIContentPolicy::TYPE_INTERNAL_SHARED_WORKER:
       return "sharedworker"_ns;
@@ -108,6 +109,7 @@
       return "empty"_ns;
     case nsIContentPolicy::TYPE_WEB_IDENTITY:
       return "webidentity"_ns;
+    case nsIContentPolicy::TYPE_END:
     case nsIContentPolicy::TYPE_INVALID:
       break;
       // Do not add default: so that compilers can catch the missing case.